pkg/config/msp.go

```diff
@@ -670,6 +670,21 @@ func (m *MSP) validateCACerts() error {
 	if err != nil {
 		return fmt.Errorf("invalid intermediate cert: %v", err)
 	}
+	rootCertPool := x509.NewCertPool()
+	for _, cert := range m.RootCerts {
+		rootCertPool.AddCert(cert)
+	}
+	opts := x509.VerifyOptions{
+		Roots: rootCertPool,
+	}
+	// Even though currently Fabric does not care if a certificate is expired or not,
+	// this cert verification will check expiration of a certificate. This will error
+	// out if the provided intermediate certs are not valid or expired.
+	for _, cert := range m.IntermediateCerts {
+		if _, err := cert.Verify(opts); err != nil {
```

Comment:
While ideally we'd use this Verify function, we can't since, as you pointed out, it considers expiration. For consistency with the other code, we should simply use CheckSignatureFrom to ensure that the intermediate certs were signed by one of the root certs.

Reply:
I just took another look at the msp code to figure out what it's doing for cert validation. I'll update the JIRA item related to this PR to include the details of what I'm thinking. 

For now, I think we go ahead with updating this PR to use cert.CheckSignatureFrom() and merge it in.  After that, we should try to incorporate cert.Verify() following the workaround that the msp code uses.

Reply:
done